### PR TITLE
Calculate correct ETag on CreateObject

### PR DESCRIPTION
**Pull request by shabbyrobe**

### CODE REVIEW

**Comment by shabbyrobe** [comment]
Hi @johannesboyne, hope you're well! I was wondering if you might be able to give this a quick once-over for me? Unfortunately, the other PR I'm working on has broadened a lot (perhaps a bit more than it ought to have 😆  ) and I was hoping to avoid the risk of merge conflicts.

**Comment by johannesboyne** [comment]
Hey @shabbyrobe, I'll tomorrow! ;-) came just back from a 1 week VAC.

**Comment by shabbyrobe** [comment]
Awesome, cheers! Hope you had a good holiday and any GitHub notifications from me did not intrude on your relaxation :smile:

File: hash.go [hunk 1]
```diff
@@ -10,17 +10,26 @@ import (
 )
 
 // hashingReader proxies an existing io.Reader, passing each read block to the
-// given hash.Hash. Once the underlying reader returns EOF, the hash is checked.
+// given hash.Hash.
+//
+// If the expected hash is not empty, once the underlying reader returns EOF,
+// the hash is checked.
```
Comment (johannesboyne) [inline hash.go]: 👍

**Pull request merged**